arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorizedArrowReader.java

```diff
@@ -407,26 +412,36 @@ public void setBatchSize(int batchSize) {
   }
 
   private static final class PositionVectorReader extends VectorizedArrowReader {
```

Comment:
We don't have to extend `VectorizedArrowReader` as this class has nothing in common but there are places in code that do a cast and it would fail otherwise.